scripts/configuration_helper.lua

```diff
@@ -86,820 +72,386 @@ function sgct.config.single(arg) end
 -- sgct.config.fisheye(msaa=1) -> 1280x720 resolution without multisampling
 function sgct.config.fisheye(arg) end
 
-function sgct.config.cube(arg) end
-
 -- Global variable storing the name of the config function called at initialization
 sgctconfiginitializeString = ""
 
+
+
+
 --[[
 ########################################################################################## 
                             Internal helper functions
 ########################################################################################## 
 ]]--
 
 function generateSingleViewportFOV(down, up, left, right, tracked)
-    return
-[[
-<Viewport ]]..tracked..[[>
-    <Pos x="0.0" y="0.0" />
-    <Size x="1.0" y="1.0" />
-    <PlanarProjection>
-        <FOV down="]]..down..[[" left="]]..left..[[" right="]]..right..[[" up="]]..up..[[" />
-        <Orientation heading="0.0" pitch="0.0" roll="0.0" />
-    </PlanarProjection>
-</Viewport>
-]]
+  local result = {}
+
+  table.insert(result,   [[            {]])
+
+  if tracked then
+    table.insert(result, [[              "tracked": ]] .. tostring(tracked) .. [[,]])
+  end
+
+  table.insert(result,   [[              "projection": {]])
+  table.insert(result,   [[                "type": "PlanarProjection",]])
+  table.insert(result,   [[                "fov": {]])
+  table.insert(result,   [[                  "left": ]] .. left .. [[,]])
+  table.insert(result,   [[                  "right": ]] .. right .. [[,]])
+  table.insert(result,   [[                  "up": ]] .. up .. [[,]])
+  table.insert(result,   [[                  "down": ]] .. down .. [[ ]])
+  table.insert(result,   [[                }]])
+  table.insert(result,   [[              }]])
+  table.insert(result,   [[            }]])
+
+  return table.concat(result, "\n")
 end
 
 
 
-function generateSingleViewport(lowerLeft, upperLeft, upperRight, tracked)
-    return
-[[
-<Viewport ]]..tracked..[[>
-    <Pos x="0.0" y="0.0" />
-    <Size x="1.0" y="1.0" />
-    <Projectionplane>
-        <Pos x="]] .. lowerLeft[1] .. [[" y="]] .. lowerLeft[2] .. [[" z="]] .. lowerLeft[3] .. [[" />
-        <Pos x="]] .. upperLeft[1] .. [[" y="]] .. upperLeft[2] .. [[" z="]] .. upperLeft[3] .. [[" />
-        <Pos x="]] .. upperRight[1] .. [[" y="]] .. upperRight[2] .. [[" z="]] .. upperRight[3] .. [[" />
-    </Projectionplane>
-</Viewport>
-]]
+function generateFisheyeViewport(fov, quality, tilt, background, crop, offset, tracked)
+  local result = {}
+
+  table.insert(result,   [[           {]])
+
+  if tracked then
+    table.insert(result, [[             "tracked": ]] .. tostring(tracked) .. ",")
```

Comment:
For consistency it could be good to choose either `[[,]]` or`","` and stick to one of them instead of mixing them.

Reply:
Done